index.js

```diff
@@ -610,36 +716,164 @@ function onStream (client, onerror) {
         client._onflushed = null
       }
 
-      next()
+      const backoffDelayMs = client._getBackoffDelay(!!err)
+      if (err) {
+        log.trace({ timeline, bytesWritten, backoffDelayMs, err },
+          'conclude intake request: error')
+        onerror(err)
+      } else {
+        log.trace({ timeline, bytesWritten, backoffDelayMs },
+          'conclude intake request: success')
+      }
+      if (backoffDelayMs > 0) {
```

Comment:
If I understand this block correctly, the `if (backoffDelayMs > 0) {` is the code that delays the handling of the next request if there's a backoff delay. 

I see in one branch of the conditional we're calling `next()` directly in the current async frame/context (calling it synchronously), but that in the other branch we're scheduling `next` to run in a different async context via `setTimeout` (calling it asynchronously). Previous versions of the code always called next synchronously. 

Proforma question -- do we have any concerns about sometimes calling `next` synchronously and other times calling it asynchronously?

Reply:
That's fair. We could put the `next()` case in a `setImmediate(next)`. I'll do that.

Aside: I suppose we could have it always do `setTimeout(next, backoffDelayMs).unref()` even if backoffDelayMs was 0. Do you have a preference? They are both nominally the same, but libuv schedules them slightly differently (https://nodejs.org/en/docs/guides/event-loop-timers-and-nexttick/).

Reply:
No strong preference or enough practical experience to say which choice would be the better choice -- happy to defer this one.